code/modules/mob/living/carbon/human/life/handle_shock.dm

```diff
@@ -2,28 +2,27 @@
 
 /mob/living/carbon/human/handle_shock()
 	..()
-	if(status_flags & GODMODE || analgesic || (species && species.flags & NO_PAIN)) return //Godmode or some other pain reducers. //Analgesic avoids all traumatic shock temporarily
+	if(status_flags & GODMODE || analgesic || (species && species.flags & NO_PAIN) || stat == DEAD || !traumatic_shock)
```

Comment:
`!traumatic_shock` means `shock_stage` could be higher than 0 and never diminish until the creature is hurt again.

Reply:
Urgh, was left over from a test.